pypeit/core/flux.py

```diff
@@ -3,209 +3,592 @@
-            for extract_type in ['boxcar', 'optimal']:
-                extract = getattr(spobj, extract_type)
-                if len(extract) == 0:
-                    continue
-#                try:
-#                    extract = getattr(spobj, extract_type)
-#                except AttributeError:
-#                    continue
-                msgs.info("Fluxing {:s} extraction for:".format(extract_type) + msgs.newline() +
-                          "{}".format(spobj))
-                wave = extract['wave']  # for convenience
-                scale = np.zeros(wave.size)
-                # Allow for some extrapolation
-                dwv = slf._sensfunc['wave_max']-slf._sensfunc['wave_min']
-                inds = ((wave >= slf._sensfunc['wave_min']-dwv*MAX_EXTRAP)
-                        & (wave <= slf._sensfunc['wave_max']+dwv*MAX_EXTRAP))
-                mag_func = utils.func_val(slf._sensfunc['c'], wave[inds],
-                                            slf._sensfunc['func'])
-                sens = 10.0**(0.4*mag_func)
-                # Extinction
-                ext_corr = extinction_correction(wave[inds], airmass, extinct)
-                scale[inds] = sens*ext_corr
-                # Fill
-                extract['flam'] = extract['counts']*scale/fitsdict['exptime'][scidx]
-                extract['flam_var'] = (extract['var']*
-                                       (scale/fitsdict['exptime'][scidx])**2)
-'''
-
-def bspline_magfit(wave, flux, var, flux_std, bspline_par=None):
+        msgs.info("Extinction correction not applied")
+
+    # Create star model
+    if (RA is not None) and (DEC is not None):
+        # Pull star spectral model from archive
+        msgs.info("Get standard model")
+        # Grab closest standard within a tolerance
+        std_dict = find_standard_file((RA, DEC))
+        # Load standard
+        load_standard_file(std_dict)
+        # Interpolate onto observed wavelengths
+        std_xspec = XSpectrum1D.from_tuple((std_dict['wave'], std_dict['flux']))
+        xspec = std_xspec.rebin(wave_star)  # Conserves flambda
+        flux_true = xspec.flux.value
+    else:
+        # Create star spectral model
+        msgs.info("Creating standard model")
+        # Create star model
+        star_loglam, star_flux, std_dict = telluric_sed(star_mag, star_type)
+        star_lam = 10 ** star_loglam
+        # Generate a dict matching the output of find_standard_file
+        std_dict = dict(file='KuruczTelluricModel', name=star_type, fmt=1,
+                        ra=None, dec=None)
+        std_dict['wave'] = star_lam * units.AA
+        std_dict['flux'] = 1e17 * star_flux * units.erg / units.s / units.cm ** 2 / units.AA
+        # ToDO If the Kuruck model is used, rebin create weird features
+        # I using scipy interpolate to avoid this
+        flux_true = scipy.interpolate.interp1d(std_dict['wave'], std_dict['flux'], fill_value='extrapolate')(wave_star)
+    
+    """
```

Comment:
remove these lines

Reply:
We will leave them prevent anyone choosing to revert to the linetools rebin which causes problems.